### ISSUE TITLE
Unexpected behaviour for events

### ISSUE THREAD

**Issue by mjmare** [Kotti/Kotti#497]
I see multiple ObjectInsert events when creating a File: 1 for kotti.filedepot.DBStoredFile and 1 for File. I expected only the latter, the one for DBStoredFile seems like an implementation detail or more suited for a separate event type. At least document this behaviour. 

When updating a file (only changing the title) I see an update event for the File (=expected), an _insert_ event for the DBStoredFile, and _another_ update event for the same File. 

I also noticed that the event handlers should not have exceptions because then one sees a non descriptive Waitress error in the browser (like something went wrong in the server). This deserves mention in the docs because it can yield errors that are hard to track down.

**Comment by tiberiuichim** [comment]
in `kotti.events._before_flush`, the following code is executed:

    for obj in session.new:
        notify(ObjectInsert(obj, req))

This means that ObjectInsert will be triggered for all new persistent records, be they of Kotti flavour or just plain SQLAlchemy ORM classes. This is the reason for the appearance of the insert event of the DBStoredFile.

I guess, if we don't want that event to be triggered for non-Kotti content types we can just filter with a `INode.providedBy(obj)` before calling `notify()`. From my perspective that's not necessary: the events can be specified for the specific context type that they want to be handled and it could be useful to have Kotti provide event machinery for SQLAlchemy plain classes, not just Kotti's.

**Comment by tiberiuichim** [comment]
I've tried manually raising errors in ObjectInsert and ObjectUpdate handlers and the tracebacks seemed very decent and legible. 

I guess the problem that mjmare highlights is Kotti's lack of default 50x pages.

**Comment by tiberiuichim** [comment]
The ObjectInsert for the DBStoredFile is caused by the `edit` method of the `kotti.views.edit.content.FileEditForm`, specifically this bit of code:

        if appstruct['file']:
            self.context.data = _to_fieldstorage(**appstruct['file'])

The problem is the use of the FileUploadTempStore. This is the class signature and docstring:

```
class FileUploadTempStore(DictMixin):
    """
    A temporary storage for file file uploads

    File uploads are stored in the session so that you don't need
    to upload your file again if validation of another schema node
    fails.
    """
```

So, to use the FileUploadTempStore, the EditForm is writing the file contents to the store (which is bad in itself, as it uses a StringIO() as a file handle, then at the end, in the form handler, it saves the file back to the database. I haven't dug deeper, but deform is overwriting the appstruct['file']['fp'] (but with the same data) even if no new file is uploaded, so apparently we'd have to compare file contents before and after, to see if file was changed.

**Comment by tiberiuichim** [comment]
The extra events problem has been fixed in #513
